### PR TITLE
(Feature) Step 3 new design

### LINKED ISSUE #1087: Apply new design to Step 3 - CrowdsaleSetup
Child of #1082

Design:

- https://zpl.io/amn7KG9 (desktop)
- https://zpl.io/a74vzXW (desktop / dutch auction)
- https://zpl.io/2j4ql94 (mobile)

### LINKED ISSUE #1153: Move plus button outside NumericInput component
See: https://github.com/poanetwork/token-wizard/pull/1148#pullrequestreview-158804506

### PR DESCRIPTION
**Pull request by gabitoesmiapodo**
**Description**

Closes #1087 
Closes #1153 

**About #1153**
 
It's just what the issue says: the _"ButtonPlus"_ component is now outside _"NumericInput"_

**About #1087** 

I think the only thing worth mentioning is that the fields order is no 100% like in the designs. This is because:

a) Some of the designs were slightly outdated, and some fields were misplaced.
b) We decided to change the order of some of the fields and controls to provide a better workflow and user experience.

Those are not a big changes, but they surely will be noticed when comparing the implementation with the original designs.

**Screenshots:**

localhost_3000_ ipad 1
localhost_3000_ ipad
localhost_3000_ laptop with mdpi screen 4
localhost_3000_ laptop with mdpi screen 3
localhost_3000_ laptop with mdpi screen 2
localhost_3000_ laptop with mdpi screen 1
localhost_3000_ ipad 2
localhost_3000_ ipad 3
localhost_3000_ iphone x 1
localhost_3000_ laptop with mdpi screen 5
localhost_3000_ iphone x
localhost_3000_ laptop with mdpi screen

### CODE REVIEW

**Comment by dennis00010011b** [comment]
@gabitoesmiapodo 
Step3: old and new styles are  mixed for fields start/end time 
latest Chrome, MacOS

screen shot 2018-10-12 at 2 07 18 pm

**Comment by dennis00010011b** [comment]
@gabitoesmiapodo 
Input fields with arrows buttons: should be disabled to set invalid values  using arrows
screen shot 2018-10-12 at 4 37 25 pm

screen shot 2018-10-12 at 4 38 10 pm

**Comment by dennis00010011b** [comment]
@gabitoesmiapodo 
Import CSV file : warning doesn't contain amount of added addresses if file contains more than 50 addresses. Warning's text should include phrase '50 addresses were added'

screen shot 2018-10-12 at 5 02 44 pm

**Comment by dennis00010011b** [comment]
@gabitoesmiapodo 
There is no placeholder for field Supply, should be 'Enter here'
screen shot 2018-10-12 at 5 23 32 pm

**Comment by dennis00010011b** [comment]
@gabitoesmiapodo 
Typo. Should be 'process'

screen shot 2018-10-12 at 5 42 13 pm

**Comment by dennis00010011b** [comment]
@gabitoesmiapodo 
Warning in console after each tier's addition

screen shot 2018-10-12 at 5 33 57 pm

**Comment by dennis00010011b** [comment]
@gabitoesmiapodo 
DutchL: there are no placeholders for input fields
screen shot 2018-10-12 at 6 13 57 pm

**Comment by gabitoesmiapodo** [comment]
@dennis00010011b 

- Step3: old and new styles are mixed for fields start/end time -> There's not really much I can do here. Native controls (arrows and such) can be disabled or hidden for Webkit browsers only, but I wouldn't recommend that. It cripples functionality.

We could use regular text inputs, but I wouldn't recommend that either: We should use date inputs for dates.

As a "fix", in addition to removing the background calendar icon when the input has focus I also removed it when the cursor hovers it. This way native controls will not overlap with the background image.

- Input fields with arrows buttons: should be disabled to set invalid values using arrows -> This should work better now.

- Import CSV file : warning doesn't contain amount of added addresses if file contains more than 50 addresses. Warning's text should include phrase '50 addresses were added' -> This is more of a programming logic issue than a styling one (which is the purpose of this issue). I created a new issue for this: https://github.com/poanetwork/token-wizard/issues/1179

- There is no placeholder for field Supply, should be 'Enter here' -> Fixed.

- Typo. Should be 'process' -> Fixed.

- Warning in console after each tier's addition -> I think this should be checked and fixed in a new issue, I created a new one here: https://github.com/poanetwork/token-wizard/issues/1180

- DutchL: there are no placeholders for input fields -> Fixed.

- Dutch - Gasprice : can't select option 'Custom' -> Ugh, I thought I got it working. It should work now.

**Comment by dennis00010011b** [comment]
@gabitoesmiapodo 
Mincap's prompt text hidden behind button
screen shot 2018-10-16 at 9 46 24 pm

**Comment by dennis00010011b** [comment]
@gabitoesmiapodo 
Minted: should be disabled to add whitelist if field supply is empty
screen shot 2018-10-16 at 10 28 55 pm

**Comment by fernandomg** [comment]
@gabitoesmiapodo despite it's specified in the redesign specs that way. Can we change the 'On/Off' wording by 'Yes/No' as in the _whitelist_ option buttons?

**Comment by fernandomg** [comment]
@gabitoesmiapodo Custom Gas Price arrows always change the decimals to `.1`

custom-gas-price_always01

File: src/components/Common/ButtonPlus.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React, { Component } from 'react'
```
Comment (fernandomg) [inline src/components/Common/ButtonPlus.js]: Same as with `ButtonCSV`

```javascript
import React from 'react'

export function ButtonPlus({ onClick, extraClassName = '' }) {
  return <div onClick={onClick} className={`sw-ButtonPlus ${extraClassName}`} />
}
```

File: src/components/Common/ButtonDelete.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React, { Component } from 'react'
```
Comment (fernandomg) [inline src/components/Common/ButtonDelete.js]: Same here. I think I forgot to save the message

```javascript
import React from 'react'

export function ButtonDelete({ onClick, extraClassName = '' }) {
  return <div onClick={onClick} className={`sw-ButtonDelete ${extraClassName}`}
}
```

File: src/components/Common/ButtonCSV.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react'
+
+export class ButtonCSV extends Component {
```
[the remark this answers is not in the document]
Comment (gabitoesmiapodo) [reply src/components/Common/ButtonCSV.js]: Shure, I will do it.

File: src/components/Common/ButtonDelete.js (hunk 2 above)
Comment (gabitoesmiapodo) [reply src/components/Common/ButtonDelete.js]: > @gabitoesmiapodo Custom Gas Price arrows always change the decimals to `.1`
> 
> custom-gas-price_always01

@fernandomg 

Ugh, this is a sh*tty one, let me see if I can make it work.

**Comment by mariano-aguero** [comment]
@gabitoesmiapodo 

can we remove this unused vars?

screenshot_20181019_100815

**Comment by gabitoesmiapodo** [comment]
@mariano-aguero 

- can we remove this unused vars? -> Yes we can.

**Pull request merged**